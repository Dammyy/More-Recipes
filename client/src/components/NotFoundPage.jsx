import React from 'react';
import { Link } from 'react-router';
import Navbar from './Navbar';
import Footer from './Footer';

export const BtnCurrent = (() => (
  'Oh no!'
));
const NotFoundPage = () => (
  <div>
    <Navbar />
    <div className="current-page btn-buttons"><BtnCurrent /></div>
    <div className="col-md-12 error-msg">
      <h1>404! Page Not Found</h1>

      <Link
        className="btn btn-primary btn-manage"
        to="/"
      >
        <i className="fa fa-home" /> Homepage
      </Link >
    </div>
    <Footer />
  </div>
);

export default NotFoundPage;
