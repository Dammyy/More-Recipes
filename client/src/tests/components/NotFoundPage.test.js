import React from 'react';
import { shallow } from 'enzyme';
import NotFoundPage from '../../components/NotFoundPage';

describe('NotFoundPage component', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const component = shallow(<NotFoundPage />);
      expect(component).toMatchSnapshot();
    });
  });
});
