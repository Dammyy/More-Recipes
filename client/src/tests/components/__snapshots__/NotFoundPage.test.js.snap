// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ReduxForm />,
        <div
          className="current-page btn-buttons"
        >
          <BtnCurrent />
        </div>,
        <div
          className="col-md-12 error-msg"
        >
          <h1>
            404! Page Not Found
          </h1>
          <Link
            className="btn btn-primary btn-manage"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            <i
              className="fa fa-home"
            />
             Homepage
          </Link>
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BtnCurrent />,
          "className": "current-page btn-buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              404! Page Not Found
            </h1>,
            <Link
              className="btn btn-primary btn-manage"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/"
            >
              <i
                className="fa fa-home"
              />
               Homepage
            </Link>,
          ],
          "className": "col-md-12 error-msg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "404! Page Not Found",
            },
            "ref": null,
            "rendered": "404! Page Not Found",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="fa fa-home"
                />,
                " Homepage",
              ],
              "className": "btn btn-primary btn-manage",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-home",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Homepage",
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ReduxForm />,
          <div
            className="current-page btn-buttons"
          >
            <BtnCurrent />
          </div>,
          <div
            className="col-md-12 error-msg"
          >
            <h1>
              404! Page Not Found
            </h1>
            <Link
              className="btn btn-primary btn-manage"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/"
            >
              <i
                className="fa fa-home"
              />
               Homepage
            </Link>
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BtnCurrent />,
            "className": "current-page btn-buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                404! Page Not Found
              </h1>,
              <Link
                className="btn btn-primary btn-manage"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/"
              >
                <i
                  className="fa fa-home"
                />
                 Homepage
              </Link>,
            ],
            "className": "col-md-12 error-msg",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "404! Page Not Found",
              },
              "ref": null,
              "rendered": "404! Page Not Found",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="fa fa-home"
                  />,
                  " Homepage",
                ],
                "className": "btn btn-primary btn-manage",
                "onlyActiveOnIndex": false,
                "style": Object {},
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-home",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Homepage",
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
